provider/rotation_data.go

```diff
@@ -0,0 +1,94 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/firehydrant/terraform-provider-firehydrant/firehydrant"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceRotation() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataFireHydrantRotation,
+		Schema: map[string]*schema.Schema{
+			// Required
+			"id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"team_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"schedule_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			// Computed
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"time_zone": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"slack_user_group_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataFireHydrantRotation(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	firehydrantAPIClient := m.(firehydrant.Client)
+
+	id := d.Get("id").(string)
+	teamID := d.Get("team_id").(string)
+	scheduleID := d.Get("schedule_id").(string)
```

Comment:
Just confirming, since these fields are required in the schema, this is a completely safe operation since we can't get to this point without a value, right?